commands/project/create.ts

```diff
@@ -28,36 +37,92 @@ exports.describe = uiBetaTag(i18n(`${i18nKey}.describe`), false);
 exports.handler = async options => {
   const { derivedAccountId } = options;
 
-  const hasCustomTemplateSource = Boolean(options.templateSource);
+  let latestRepoReleaseTag;
+  let templateSource = options.templateSource;
+  if (!templateSource) {
+    templateSource = HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH;
+    try {
+      const releaseData = await fetchReleaseData(
+        HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH
+      );
+      if (releaseData) {
+        latestRepoReleaseTag = releaseData.tag_name;
+      }
+    } catch (err) {
+      logger.error(i18n(`${i18nKey}.error.failedToFetchProjectList`));
+      process.exit(EXIT_CODES.ERROR);
+    }
+  }
 
-  let githubRef = '';
+  const projectTemplates = await getProjectTemplateListFromRepo(
+    templateSource,
+    latestRepoReleaseTag || DEFAULT_PROJECT_TEMPLATE_BRANCH
+  );
 
-  if (!hasCustomTemplateSource) {
-    const releaseData = await fetchReleaseData(
-      HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH
-    );
-    githubRef = releaseData.tag_name;
+  if (!projectTemplates.length) {
+    logger.error(i18n(`${i18nKey}.error.failedToFetchProjectList`));
+    process.exit(EXIT_CODES.ERROR);
   }
 
   const { name, template, dest } = await createProjectPrompt(
-    githubRef,
-    options
+    options,
+    projectTemplates
   );
+  const projectName = name || options.name;
```

Comment:
Same as my above comment - looks like `name` and `dest` are guaranteed to be returned